### PR TITLE
CI: automagically merge approved PRs

### PR DESCRIPTION
**Pull request by SMillerDev**
After making all changes to the cask:

- [ ] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.
- [ ] The commit message includes the cask’s name and version.
- [ ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not [already refused].
- [ ] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

File: .github/workflows/autopublish.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Publish and commit bottles on schedule
```
Comment (MikeMcQuaid) [inline .github/workflows/autopublish.yml:1]:
```suggestion
name: Publish and commit casks on schedule
```

File: .github/workflows/autopublish.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    # Once every hour
+    - cron: "*/60 * * * *"
+
+concurrency:
+  group: autopublish
+  cancel-in-progress: true
+
+env:
+  HOMEBREW_FORCE_HOMEBREW_ON_LINUX: 1
+
```
Comment (MikeMcQuaid) [inline .github/workflows/autopublish.yml:15]:
```suggestion
```

File: .github/workflows/publish-commit-bottles.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
+name: Publish and commit bottles
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-bottles.yml:1]:
```suggestion
name: Publish casks
```

and rename the file too, I reckon.

File: .github/workflows/publish-commit-bottles.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    inputs:
+      pull_request:
+        description: Pull request number
+        required: true
+      args:
+        description: "Extra arguments to `brew pr-pull`"
+        default: ""
+      self_hosted:
+        description: "Whether to run the upload job on a self-hosted runner (default: false)"
+        required: false
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-bottles.yml:14]:
```suggestion
```

File: .github/workflows/publish-commit-bottles.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+env:
+  GNUPGHOME: /tmp/gnupghome
+  HOMEBREW_DEVELOPER: 1
+  HOMEBREW_NO_AUTO_UPDATE: 1
+
+jobs:
+  upload:
+    runs-on: ${{github.event.inputs.self_hosted == 'true' && 'linux-self-hosted-1' || 'ubuntu-latest'}}
+    container: ${{github.event.inputs.self_hosted == 'true' && fromJson('{"image":"ghcr.io/homebrew/ubuntu16.04:master","options":"--user=linuxbrew -e GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED"}') || ''}}
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-bottles.yml:24]: Don't need all this custom logic

File: .github/workflows/autopublish.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  autopublish:
+    if: github.repository == 'Homebrew/homebrew-core'
```
Comment (bevanjkay) [inline .github/workflows/autopublish.yml:18]: Does this need to be changed to match the cask repos?

**Comment by SMillerDev** [comment]
Thanks both, I tried to keep it close to the version in Homebrew/core, but that might have been a little too close in hindsight.

File: .github/workflows/autopublish.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+name: Publish and publish casks on schedule
```
Comment (bevanjkay) [inline .github/workflows/autopublish.yml:1]:
```suggestion
name: Publish casks on schedule
```

File: .github/workflows/publish-commit-casks.yml [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+      - name: Dismiss approvals on failure
+        if: ${{!success()}}
+        uses: Homebrew/actions/dismiss-approvals@master
+        with:
+          token: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
+          pr: ${{github.event.inputs.pull_request}}
+          message: "bottle publish failed"
+
+      # Workaround until the `cache` action uses the changes from
+      # https://github.com/actions/toolkit/pull/580.
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-casks.yml:109]: This PR was rejected so we should probably stop using these workarounds. Could you make an issue to track removing these?

File: .github/workflows/publish-commit-casks.yml [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+          BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
+          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
+          HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_CORE_PUBLIC_REPO_EMAIL_TOKEN}}
+        run: brew pr-pull --debug --workflows=ci.yml --committer="$BREWTESTBOT_NAME_EMAIL" --tap homebrew/cask ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
+
+      - name: Push commits
+        uses: Homebrew/actions/git-try-push@master
+        with:
+          token: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
+          directory: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-casks.yml:84]:
```suggestion
          directory: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask
```

File: .github/workflows/publish-commit-casks.yml (hunk 8 above)
Comment (SMillerDev) [reply .github/workflows/publish-commit-casks.yml:109]: Here or in core where I copied it from?

Comment (MikeMcQuaid) [reply .github/workflows/publish-commit-casks.yml:109]: Core. Fixing doesn't need to block this PR but opening the issue should.

Comment (SMillerDev) [reply .github/workflows/publish-commit-casks.yml:109]: Done in: https://github.com/Homebrew/homebrew-core/issues/98676

**Review by carlocab — COMMENTED** [review]
I understand the value of not deviating much from the equivalent homebrew-core workflow, but I wonder if it's not simpler to just run `gh pr merge --squash` on approved PRs?

**Comment by SMillerDev** [comment]
I like the autosquash consistency here, but I wouldn't mind that approach either. I just couldn't imagine a setup that would allow this on a schedule.

**Review by MikeMcQuaid — APPROVED** [review]

File: .github/workflows/autopublish.yml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    # Once every hour
+    - cron: "*/60 * * * *"
+
+concurrency:
+  group: autopublish
+  cancel-in-progress: true
+
+jobs:
+  autopublish:
+    if: github.repository == 'Homebrew/homebrew-cask'
```
Comment (bevanjkay) [inline .github/workflows/autopublish.yml:15]: @SMillerDev Do we want this to behaviour to be mirrored across the other cask repos? Can we use a wildcard in the workflow here for `homebrew-cask-*`?

Comment (SMillerDev) [reply .github/workflows/autopublish.yml:15]: I think we can start with the current approach and then we could check for regexes.

**Pull request merged**